### DISCUSSION TITLE
Trouble getting TDOA values from Loco Positioning system

### DISCUSSION THREAD

**Discussion by flapjack48** [bitcraze/discussions#110] [Q&A] [answered]
Hello,

I recently purchased the Loco Positioning systems to add onto an already established ROS environment with motion capture and some upgraded (brushless) Crazyflies with the BigQuad deck. We bought the Loco system to access the TDOA information as we are researching RF localization. I set up my anchors according to the "Getting started with the Loco Positioning system" from https://www.bitcraze.io/documentation/tutorials/getting-started-with-loco-positioning-system/
I was successful in updating, configuring and finding all the anchors and the calibration seemed successful based on moving the Crazyflie around the environment. I switched the system to TDOA3 (as that was recommended as the more robust option for TDOA) and that appeared to work from the cfclient with green boxes on all anchors. 

With that all working, I added the logging variable, tdoaEngine.tdoa to the Crazyflie using the cfclient and wrote a Python script according to the logging variables Bitcraze page. Attached below:
[tdoa_log.txt](https://github.com/bitcraze/discussions/files/8952912/tdoa_log.txt)

But my output is just "0" seen here:
![Image](https://user-images.githubusercontent.com/68208005/174900967-02d18a14-b728-47da-b8b0-46461bc0ff79.jpeg)

I found the logging variable tdoa3.tdoa from the "Logging" section of "TDoA2 VS TDoA3" (https://www.bitcraze.io/documentation/repository/lps-node-firmware/master/functional-areas/tdoa2-vs-tdoa3/)
But for some reason that variable is not available to add from the cfclient so when I use that in my code I just error out seen here:
![Image](https://user-images.githubusercontent.com/68208005/174901735-320a1de6-4bb3-4b04-a2a8-21daceced973.jpeg)

Does anyone have any ideas on what I am doing wrong for either method? Has anyone had success accessing the tdoa values and if so which one? 
Again, to reiterate I don't really care how I get the tdoa value or which method I use, even using TOF if necessary. 

Note: I did update both the Crazyflie firmware and the cfclient today as well.

Any help is much appreciate!!
Sasha

**Comment by krichardsson** [comment]
Hi!

The logging in the TDoA engine supports logging data for one anchor pair at a time. You chose which anchor pair to log by setting the parameters [tdoaEngine.logId](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/params/#tdoaenginelogid) and [tdoaEngine.logOthrId](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/params/#tdoaenginelogothrid) to the ids of the two anchors. You will get the TDoA in the [tdoaEngine.tdoa](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/logs/#tdoaenginetdoa) log and the TOF between the anchors in [tdoaEngine.tof](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/logs/#tdoaenginetof).

For example, if you are interested in the anchor pair with ids 3 and 7, set tdoaEngine.logId=3 and tdoaEngine.logOthrId=7. After this you will get the TDoA and TOF values in the log variables mentioned above.

**Reply by flapjack48** [reply]
Hello,

Thanks for getting back to me so quickly. Although I agree that you would need to set the anchor pairs, they default to ids 0 and 1 so you can test everything else without setting the ids. 
For completeness the problem was resolved and came from not having "logconf.start()" under the with statement

**Reply by knmcguire** [reply]
@flapjack48 could you select kristoffer's answer as the answer to your question? or if it is not complete, suggest an answer here below and select that? This way the answer to your issue will be listed on top and it is easier for other users to find it. 

Thanks!

**Comment by flapjack48** [comment, marked as answer]
I needed to add "logconf.start()" under the with statement shown in the file attached
[tdoa_log.txt](https://github.com/bitcraze/discussions/files/9202911/tdoa_log.txt)

**Comment by bhavya1600** [comment]
Was this code to get the current drone position? @flapjack48
